src/stdlib_strings.f90

```diff
@@ -290,5 +299,72 @@ elemental function ends_with_string_string(string, substring) result(match)
 
     end function ends_with_string_string
 
+    !> Extract the characters from the region between 'first' and 'last' index (both inclusive)
+    !> of the input 'string' by taking strides of length 'stride'
+    !> Returns a new string
+    elemental function slice_string(string, first, last, stride) result(sliced_string)
```

Comment:
See https://github.com/fortran-lang/stdlib/issues/413#issuecomment-846933463 for the discussion why elemental can't work here.

Reply:
Yeah, totally understand why it can't be elemental for `character`. I was more referring to the discussions in #319 that would also apply to why we may not want `slice _string` to be elemental. But I think the resolution was that we should allow it to be elemental, and I agree with that. Just thought I'd bring it up in case sentiments may have changed or there was some nuance here that would sway the decision the other way.